### PR TITLE
Fix python generation when custom files and templates are specified

### PR DESCRIPTION
**Pull request by shylasrinivas**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by shylasrinivas** [comment]
@taxpon  @frol  @mbohlool  @cbornet  @kenjones-cisco  @tomplus  @Jyhess  @arun-nalla  @spacether

**Comment by shylasrinivas** [comment]
Fix for https://github.com/OpenAPITools/openapi-generator/issues/9570

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java [hunk 1]
```diff
@@ -938,7 +938,7 @@ private void processUserDefinedTemplates() {
         // TODO: initial behavior is "merge" user defined with built-in templates. consider offering user a "replace" option.
         if (userDefinedTemplates != null && !userDefinedTemplates.isEmpty()) {
             Map<String, SupportingFile> supportingFilesMap = config.supportingFiles().stream()
-                    .collect(Collectors.toMap(TemplateDefinition::getTemplateFile, Function.identity()));
+                    .collect(Collectors.toMap(TemplateDefinition::getTemplateFile, Function.identity(), (oldValue, newValue) -> oldValue));
```
Comment (spacether) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java:941]: What is this lambda doing? Why is it returning the oldValue?
I have no way to tell from this PR if this code works or not.
If this feature is changed in the future, we will not have any test that fails and the feature will break.
How about adding a test of processUserDefinedTemplates to ensure that this feature will keep working?

Comment (shylasrinivas) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java:941]: @spacether Absolutely valid points. 
Collectors.toMap throws IllegalStateException when dup keys are found, hence the logic is to return a single value when dups are encountered. oldValue is an existing key in the map, whereas newValue is a key found during later iterations of the list (in this case config.supportingFiles()). The code simply retains the existing key (oldValue) in the map when a dup (newValue) is encountered. Based on my testing, I have found oldValue to be user defined templates, which is desired to be retained in the map.
I will look into adding a test. Agreed there is no way to tell if the code works, how do you typically ensure PR quality?

Comment (spacether) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java:941]: Thanks for explaining what this code is doing.
Most of our most basic tests are in: https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/java/org/openapitools/codegen/DefaultCodegenTest.java

Comment (shylasrinivas) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java:941]: @spacether Thank you for the pointers. I have added a test, but it looks like the config (line 684 of DefaultCodegenTest.java) I want to specify is not supported in CodegenConfigurator unless, I am missing something. Lmk if the test looks ok. I can remove the commented check based on your input.

Comment (spacether) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java:941]: Thank you for explaining that and adding the test. Your test looks good.

**Review by spacether — APPROVED** [review]
Thank you for your PR. This looks good

**Pull request merged**